### ISSUE TITLE
Item search checkboxes toggling cause crash

### ISSUE THREAD

**Issue by Silentguard** [domialex/Sidekick#361]
How to reproduce:
Fast toggle random checkboxes during search process

```
2020-04-24 22:26:11.188 +08:00 [INF] Fetching Trade API Listings from Query XLWw3K4sP.
2020-04-24 22:26:11.278 +08:00 [FTL] Unhandled exception in application root
System.NullReferenceException: Object reference not set to an instance of an object.
   at Sidekick.Views.Prices.PriceViewModel.LoadMoreData()
   at Sidekick.Views.Prices.PriceViewModel.UpdateQuery()
   at Sidekick.Views.Prices.PriceViewModel.<InitializeFilter>b__48_0[T](Object sender, PropertyChangedEventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
2020-04-24 22:26:11.475 +08:00 [FTL] Unhandled exception in application root
System.NullReferenceException: Object reference not set to an instance of an object.
   at Sidekick.Views.Prices.PriceViewModel.LoadMoreData()
   at Sidekick.Views.Prices.PriceViewModel.UpdateQuery()
   at Sidekick.Views.Prices.PriceViewModel.<InitializeFilter>b__48_0[T](Object sender, PropertyChangedEventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by domialex** [comment]
Does this happen all the time from on your end? It seems your Sidekick has a little problem, did you try re-downloading the [latest version](https://github.com/domialex/Sidekick/releases/latest/download/Sidekick.exe)?

**Comment by Silentguard** [comment]
I have latest version. I tried to reproduce another issue, when toggling checkbox cause massive OS freeze sometimes but got this instead

**Comment by leMicin** [comment]
This dhould be fixed with the changes coming with #371 . Checking and unchecking checkboxes will not trigger the refresh immediately, there will be a debounce.
